File: bench/kernels/bench_fib.go
// bench_fib.go - recursive Fibonacci, n=38
package main

import (
	"fmt"
	"time"
)

func fib(n int) int64 {
	if n <= 1 {
		return int64(n)
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	n := 38
	start := time.Now()
	result := fib(n)
	ms := time.Since(start).Milliseconds()
	fmt.Printf("result: %d\nelapsed: %d ms\n", result, ms)
}

File: bench/kernels/bench_fib.py
# bench_fib.py - recursive Fibonacci, n=38
import time, sys
sys.setrecursionlimit(100_000)

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

n = 38
t0 = time.monotonic()
result = fib(n)
elapsed = int((time.monotonic() - t0) * 1000)
print(f"result: {result}")
print(f"elapsed: {elapsed} ms")
